### DISCUSSION TITLE
Why does zeekygen need to run `zeek_init()`?

### DISCUSSION THREAD

**Discussion by ckreibich** [zeek/zeek#4130] [General]
This is a follow-up to brief conversation in #4120 around how to run Zeek for Zeekygen.

The way Zeekygen currently operates is "in passing" at startup: its manager traces script parsing, after the post-init phase we ask it to generate the docs, and then Zeek initializes and runs as usual except the `zeek_init()` in `scripts/zeekygen/__load__.zeek` calls `terminate()` to stop Zeek.

I am wondering: for generating the docs, why do we need to run `zeek_init()` at all? I just changed the Zeek invocation in `ci/update-zeekygen-docs.sh` to quit after parsing ...
```
--- i/ci/update-zeekygen-docs.sh
+++ w/ci/update-zeekygen-docs.sh
@@ -28,7 +28,7 @@ cd $build_dir
 export ZEEK_SEED_FILE=$source_dir/testing/btest/random.seed

 function run_zeek {
-    ZEEK_ALLOW_INIT_ERRORS=1 zeek -X $conf_file zeekygen >/dev/null
+    ZEEK_ALLOW_INIT_ERRORS=1 zeek -a -X $conf_file zeekygen >/dev/null

     if [ $? -ne 0 ]; then
         echo "Failed running zeek with zeekygen config file $conf_file" >&2
```
... and it leads to identical updates to the docs. Seems to me this would avoid any side-effects out of `zeek_init()` for purposes of generating docs, but maybe I'm missing something docs-relevant that can happen at that stage still. Thoughts?

**Comment by awelzel** [comment]
> I am wondering: for generating the docs, why do we need to run zeek_init() at all? I just changed the Zeek invocation in ci/update-zeekygen-docs.sh to quit after parsing ...

Nice! I think that makes total sense :-)

For reference, we previously swallowed stderr, so warnings weren't such a big deal unless there was an actual failure:
https://github.com/zeek/zeek/pull/3363/files

**Comment by ckreibich** [comment]
This landed with zeek/zeek#4551, so I'm closing the discussion.

 (I guess "resolved" does not apply, and the mentioning of this discussion over in that PR also doesn't show up here. Hrrm.)
